core/foundation/inc/ROOT/RConfig.hxx

```diff
@@ -292,10 +292,14 @@
 #   if defined(__i386__)
 #      define R__BYTESWAP
 #   endif
-#   if defined(__arm__) || defined (__arm64__)
+#   if defined(__x86_64__)
 #      define R__BYTESWAP
+#      define R__B64      /* enable when 64 bit machine */
 #   endif
-#   if defined(__x86_64__)
+#   if defined(__arm__)
+#      define R__BYTESWAP
+#   endif
+#   if defined (__arm64__)
```

Comment:
Are `__arm__` and `__arm64__` exclusive? If we don't know show sure maybe consider
```suggestion
#   if defined(__arm__) || defined (__arm64__)
#      define R__BYTESWAP
#   endif
#   if defined (__arm64__)
#      define R__B64
#   endif
```

Reply:
I believe they are exclusive, see e.g. how we use them [here](https://github.com/root-project/root/blob/master/core/unix/src/TUnixSystem.cxx#L932-L935), it seems that `__arm64__` does not imply `__arm__`.